### PR TITLE
Updated Ktor to 2.0.0-beta-1, Kotlin to 1.6.10, Coroutines to 1.6.0

### LINKED ISSUE #13: Kamel with Ktor 2.0.0-beta-1 triggers a NoClassDefFoundError for Desktop app
Hello,

I was using Kamel with Ktor 2.0.0-beta-1 and when building the Desktop client I've had this error:
> Reason=java.lang.NoClassDefFoundError: io/ktor/client/features/BodyProgressKt

This seems to happen due to the different packaging that Ktor had when upgrading to the major version 2.0.

I've fixed it locally, and there's an upcoming PR for it 🙂

**Edit:** I've updated the PR to ktor 2.0.0

### PR DESCRIPTION
**Pull request by cmota**
This PR aims to fix #13.

While making this change I've also updated all the deprecated libraries 🙂 .

### CODE REVIEW

**Comment by Syer10** [comment]
You could update this to Ktor 2.0.0 now that its released

File: kamel-core/src/commonMain/kotlin/io/kamel/core/fetcher/HttpFetcher.kt [hunk 1]
```diff
@@ -29,15 +28,15 @@ internal class HttpFetcher(private val client: HttpClient) : Fetcher<Url> {
...
+        val response = client.request {
             onDownload { bytesSentTotal, contentLength ->
                 val progress = (bytesSentTotal.toFloat() / contentLength).coerceAtMost(1.0F)
                 send(Resource.Loading(progress))
             }
             takeFrom(resourceConfig.requestData)
             url(data)
         }
-        val bytes = response.receive<ByteReadChannel>()
+        val bytes = response.body<ByteReadChannel>()
```
Comment (Syer10) [inline kamel-core/src/commonMain/kotlin/io/kamel/core/fetcher/HttpFetcher.kt:39]: Tested this in Ktor 2.0.0, it needs to be
```suggestion
        val bytes = response.bodyAsChannel()
```

**Comment by cmota** [comment]
Thank you for the heads up @Syer10! I've just updated to ktor 2.0.0 and AGP 7.1.3.

I also made your suggestion on `HttpFetcher` 🙂

**Comment by dragossusi** [comment]
Is this repository still maintained?

**Comment by cmota** [comment]
@dragossusi I'm using the branch that I've created with the Ktor 2.0.0 support and I'll keep updating it until another solution is around. I'm using it in a couple of apps, without any problem.

Unfortunately, I don't think the original author is maintaining it.

**Comment by dragossusi** [comment]
@cmota did you upload it to maven? I was thinking about creating a new image loading library like this one and make it available for all platforms(I'll make it use dev builds of `compose-jb`)

**Comment by cmota** [comment]
I didn't because only the author can publish the library. What I'm doing is using it locally.

**Comment by luca992** [comment]
@cmota I found a bunch of issues with your fork causing the build to fail. I fixed them and further updated ktor and some plugins. I'm going to publish a different branch updated to kotlin 1.6.21 and the latest dev compose to a maven repo for anyone interested. I'll post it here when I'm done

**Comment by alialbaali** [comment]
Thanks everyone for their support. If you'd like to contribute to this project, please let me know so I can add you to the project as a collaborator.

**Comment by cmota** [comment]
Hello @luca992, can you share the issue? I've been using this without any problem for quite some time 🤔.

**Comment by luca992** [comment]
If you build as is:
1. `gradlePluginPortal()` isn't added to `buildscript.repositories` and it fails ther
2. Build fails if extra publishing properties aren't present (which isn't necessarily a bug, but it's easy enough to fix by doing ` as String` -> ` as String? ?: ""`
3. The error: Minimum supported Gradle version is 7.2. Current version is 7.1.1. (I updated to 7.4.2)
4. `bodyAsChannel()` uses the wrong import in `HttpFetcher`
5. 
```
> Task :kamel-image:compileDebugKotlinAndroid FAILED
e: /Users/lucaspinazzola/Projects/Kamel/kamel-image/src/androidMain/kotlin/io/kamel/image/fetcher/ResourcesFetcher.kt: (25, 12): Unresolved reference: ExperimentalIoApi
e: /Users/lucaspinazzola/Projects/Kamel/kamel-image/src/androidMain/kotlin/io/kamel/image/fetcher/ResourcesFetcher.kt: (25, 12): An annotation argument must be a compile-time constant
e: /Users/lucaspinazzola/Projects/Kamel/kamel-image/src/androidMain/kotlin/io/kamel/image/mapper/ResourcesIdMapper.kt: (10, 73): This annotation is not applicable to target 'type usage'
e: /Users/lucaspinazzola/Projects/Kamel/kamel-image/src/androidMain/kotlin/io/kamel/image/mapper/ResourcesIdMapper.kt: (16, 68): Cannot find a parameter with this name: encodedPath
```
6. android gradle plugin `7.1.3` fails with a strange must use a different android studio version.. I downgraded back to `7.0.4`, which is what all `compose-jb` examples use

**Comment by luca992** [comment]
I published a release of my fork here if anyone is waiting for this to be merged:
```
implementation("io.github.luca992.com.alialbaali.kamel:kamel-image:0.3.0-ktor2")
```

**Pull request merged**